src/Bicep.LangServer/BicepParamsCompilationManager.cs

```diff
@@ -50,13 +49,17 @@ public void UpsertCompilation(DocumentUri uri, int? version, string text, string
         {
             var paramsFile = SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text);
 
-            var getCompilation = (Uri uri) => {
-                var bicepConfig = bicepConfigurationManager.GetConfiguration(uri);
-                var bicepCompilationContext = bicepCompilationContextProvider.Create(new Workspace(), uri, new Dictionary<ISourceFile, ISemanticModel>().ToImmutableDictionary(), bicepConfig, new LinterAnalyzer(bicepConfig));
-                return bicepCompilationContext.Compilation;
-            };
+            Uri? bicepFileUri = PathHelper.TryGetBicepFileUri(out var diagnosticWriter, fileResolver, paramsFile);
+            var compilationLoadDiagnostics = diagnosticWriter.GetDiagnostics().ToImmutableArray();
+
+            var semanticModel = new ParamsSemanticModel(paramsFile, compilationLoadDiagnostics);
```

Comment:
This logic is repeated in the CLI and LS but I think is fine since it prevents me having to worry about making LS functions ```async```